### PR TITLE
Add option to disable cloudwatch dashboards

### PR DESCRIPTION
**Pull request by mwiarda**
Hi 👋 

Thanks for your awesome work!

I added the option to disable dashboards as requested in #3

### CODE REVIEW

File: src/watchful.ts [hunk 1]
```diff
@@ -51,6 +51,13 @@ export interface WatchfulProps {
    * @default []
    */
   readonly alarmActionArns?: string[];
+
+  /**
+   * If set to true then no cloudwatch dashboards are generated
+   *
+   * @default false
+   */
+  readonly noDashboard?: boolean;
```
Comment (BenChaimberg) [inline src/watchful.ts:60]: switch this to a positive. also please move it next to the dashboardName prop just for organizational purposes
```suggestion
   * Whether to generate CloudWatch dashboards
   *
   * @default true
   */
  readonly dashboard?: boolean;
```

File: src/watchful.ts [hunk 2]
```diff
@@ -86,15 +93,18 @@ export class Watchful extends Construct implements IWatchful {
       );
     }
 
-    this.dash = new cloudwatch.Dashboard(this, 'Dashboard', { dashboardName: props.dashboardName });
+    if (!props.noDashboard){
```
Comment (BenChaimberg) [inline src/watchful.ts:96]: should also trigger if dashboardName is provided (requires a doc update)

File: src/watchful.ts [hunk 3]
```diff
@@ -86,15 +93,18 @@ export class Watchful extends Construct implements IWatchful {
...
+      this.dash = new cloudwatch.Dashboard(this, 'Dashboard', { dashboardName: props.dashboardName });
```
Comment (BenChaimberg) [inline src/watchful.ts:97]: the dash property in this class should be explicitly marked as undefined (`private dash?: cloudwatch.Dashboard`)

**Review by BenChaimberg — CHANGES_REQUESTED** [review]

File: src/watchful.ts (hunk 2 above)
Comment (mwiarda) [reply src/watchful.ts]: Changed condition to:
```
const dashboard = props.dashboard ?? true;
if (dashboard || props.dashboardName){
```
and updated the docs.

File: src/watchful.ts [hunk 4]
```diff
@@ -86,15 +95,19 @@ export class Watchful extends Construct implements IWatchful {
       );
     }
 
-    this.dash = new cloudwatch.Dashboard(this, 'Dashboard', { dashboardName: props.dashboardName });
+    const dashboard = props.dashboard ?? true;
+    if (dashboard || props.dashboardName){
```
Comment (BenChaimberg) [inline src/watchful.ts:99]: Sorry, I confused myself a bit with the double negatives in the last review. This should actually throw an error if `dashboard` is `false` and `dashboardName` is provided
```suggestion
    if (props.dashboard === false && props.dashboardName) {
      throw new Error('Dashboard name is provided but dashboard creation is disabled');
    }
    if (props.dashboard !== false) {
```

File: src/watchful.ts [hunk 5]
```diff
@@ -40,6 +40,8 @@ export interface WatchfulProps {
 
   /**
    * The name of the CloudWatch dashboard generated by Watchful.
+   * Overrides dashboard property if specified.
+   *
```
Comment (BenChaimberg) [inline src/watchful.ts:44]:
```suggestion
```

**Review by BenChaimberg — CHANGES_REQUESTED** [review]

File: src/watchful.ts [hunk 6]
```diff
@@ -40,6 +40,7 @@ export interface WatchfulProps {
 
   /**
    * The name of the CloudWatch dashboard generated by Watchful.
+   *
```
Comment (BenChaimberg) [inline src/watchful.ts:43]:
```suggestion
```

File: src/watchful.ts [hunk 7]
```diff
@@ -51,10 +52,17 @@ export interface WatchfulProps {
    * @default []
    */
   readonly alarmActionArns?: string[];
+
+  /**
+   * Whether to generate CloudWatch dashboards
+   *
```
Comment (BenChaimberg) [inline src/watchful.ts:58]:
```suggestion
```

**Review by BenChaimberg — CHANGES_REQUESTED** [review]

**Comment by BenChaimberg** [comment]
please pull and rebuild, then should be good to go! thanks!

**Comment by mwiarda** [comment]
Sorry. Build fails because GitHub actions does not have the necessary permissions. 
Do you have any pointers on what I have to do to get that fixed?

**Comment by BenChaimberg** [comment]
It's attempting to commit changes because there are changes that occur during packaging. I was asking you to `yarn build` in your local workspace and add any changes

**Review by BenChaimberg — APPROVED** [review]

**Pull request merged**